Library/Homebrew/rubocops/keg_only.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+require "rubocops/extend/formula"
+
+module RuboCop
+  module Cop
+    module FormulaAudit
+      class KegOnly < FormulaCop
+        def audit_formula(node, _class_node, _parent_class_node, body_node)
+          keg_only_node = find_node_method_by_name(body_node, :keg_only)
+          return unless keg_only_node
+
+          offending_node(node)
+
+          whitelist = %w[
+            Apple
+            macOS
+            OS
+            Homebrew
+            Xcode
+            GPG
+            GNOME
+            BSD
+            Firefox
+          ].freeze
+
+          reason = string_content(parameters(keg_only_node).first)
+          # formula_name = find_node_method_by_name(body_node, :name)
```

Comment:
Should these lines be moved to `audit` or is a RuboCop check enough? Slightly confused here, hence commented these lines out for now.